lang/cs/Org.Apache.REEF.Network/Group/Pipelining/PipelineMessageCodec.cs

```diff
@@ -0,0 +1,71 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Wake.Remote;
+using System;
+using System.Linq;
+
+namespace Org.Apache.REEF.Network.Group.Pipelining
+{
+    /// <summary>
+    /// The codec for PipelineMessage
+    /// </summary>
+    /// <typeparam name="T">The message type</typeparam>
+    public class PipelineMessageCodec<T> : ICodec<PipelineMessage<T>>
```

Comment:
I am thinking if we really need pipelining folder. If not, putting PipelineMessageCodec into Codec folder would make more sense.  And PipelineMessage can go together with GroupCommunicationMessage.